File: prime.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <stdbool.h>

bool is_prime(long n) {
    printf("Checking %ld for primality...\n", n);
    if (n <= 1) return false;
    if (n <= 3) return true;
    if (n % 2 == 0 || n % 3 == 0) return false;

    long i = 5;
    while (i * i <= n) {
        printf("Testing %ld and %ld\n", i, i+2);
        if (n % i == 0 || n % (i+2) == 0) return false;
        i += 6;
    }
    return true;
}

void generate_primes(long limit) {
    printf("Generating primes up to %ld\n", limit);
    for (long i = 2; i <= limit; i++) {
        if (is_prime(i)) {
            printf("%ld ", i);
        }
    }
    printf("\n");
}

long next_prime(long n) {
    printf("Finding next prime after %ld\n", n);
    long candidate = n + 1;
    while (!is_prime(candidate)) {
        candidate++;
    }
    return candidate;
}

void prime_factors(long n) {
    printf("Prime factors of %ld: ", n);
    while (n % 2 == 0) {
        printf("2 ");
        n /= 2;
    }
    long i = 3;
    while (i * i <= n) {
        while (n % i == 0) {
            printf("%ld ", i);
            n /= i;
        }
        i += 2;
    }
    if (n > 2) printf("%ld", n);
    printf("\n");
}

int main() {
    int choice;
    long n;
    while (1) {
        printf("\n1. Check prime\n2. Generate primes\n3. Next prime\n4. Prime factors\n5. Exit\n");
        printf("Enter choice: ");
        scanf("%d", &choice);
        switch(choice) {
            case 1:
                printf("Enter number: ");
                scanf("%ld", &n);
                printf("%ld is %sprime\n", n, is_prime(n) ? "" : "not ");
                break;
            case 2:
                printf("Enter limit: ");
                scanf("%ld", &n);
                generate_primes(n);
                break;
            case 3:
                printf("Enter number: ");
                scanf("%ld", &n);
                printf("Next prime: %ld\n", next_prime(n));
                break;
            case 4:
                printf("Enter number: ");
                scanf("%ld", &n);
                prime_factors(n);
                break;
            case 5:
                exit(0);
            default:
                printf("Invalid choice\n");
        }
    }
}

File: prime.py
import math
import sys
import random
from functools import reduce

print("Welcome to the Over-engineered Prime Utility in Python!")

def is_prime(n):
    print(f"Checking if {n} is prime...")
    if n <= 1:
        print(f"{n} <= 1, so not prime")
        return False
    if n <= 3:
        print(f"{n} <= 3, so prime")
        return True
    if n % 2 == 0 or n % 3 == 0:
        print(f"{n} divisible by 2 or 3, not prime")
        return False
    i = 5
    while i * i <= n:
        print(f"Testing divisibility by {i} and {i+2}")
        if n % i == 0 or n % (i + 2) == 0:
            return False
        i += 6
    return True

def generate_primes(limit):
    print(f"Generating primes up to {limit}...")
    primes = []
    for n in range(2, limit+1):
        if is_prime(n):
            primes.append(n)
    return primes

def next_prime(n):
    print(f"Finding next prime after {n}")
    candidate = n + 1
    while True:
        if is_prime(candidate):
            return candidate
        candidate += 1

def prime_factors(n):
    print(f"Factorizing {n} into prime factors...")
    factors = []
    while n % 2 == 0:
        factors.append(2)
        n //= 2
    i = 3
    while i*i <= n:
        while n % i == 0:
            factors.append(i)
            n //= i
        i += 2
    if n > 2:
        factors.append(n)
    return factors

def redundant_prints(limit):
    print("Let's print a lot of unnecessary numbers...")
    for i in range(limit):
        if i % 2 == 0:
            print(f"{i} is even")
        else:
            print(f"{i} is odd")
    print("Done printing!")

# Main program
while True:
    print("\nChoose an option:")
    print("1. Check prime")
    print("2. Generate primes")
    print("3. Next prime")
    print("4. Prime factors")
    print("5. Print redundant info")
    print("6. Exit")

    choice = input("Enter choice: ")
    if choice == "1":
        n = int(input("Enter number: "))
        print(is_prime(n))
    elif choice == "2":
        n = int(input("Enter limit: "))
        print(generate_primes(n))
    elif choice == "3":
        n = int(input("Enter number: "))
        print(next_prime(n))
    elif choice == "4":
        n = int(input("Enter number: "))
        print(prime_factors(n))
    elif choice == "5":
        n = int(input("Enter a number: "))
        redundant_prints(n)
    elif choice == "6":
        print("Exiting...")
        break
    else:
        print("Invalid choice")
